src/rust/event-source/src/server.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+
+use rust_proto_new::{
+    graplinc::grapl::api::event_source::{
+        v1beta1 as native,
+        v1beta1::server::{
+            EventSourceApi,
+            EventSourceServer,
+        },
+    },
+    protocol::healthcheck::HealthcheckStatus,
+};
+use tokio::net::TcpListener;
+
+use crate::{
+    config::EventSourceConfig,
+    db::EventSourceDbClient,
+    error::EventSourceError,
+};
+
+pub async fn exec_service(config: EventSourceConfig) -> Result<(), Box<dyn std::error::Error>> {
+    let api_impl = EventSourceApiImpl::try_from(&config).await?;
+    // I haven't quite figured out the right place to put the migrate.
+    // Perhaps when we construct the db client?!
+    api_impl.db_client.migrate().await?;
```

Comment:
I think you want to use the `migrate!` macro

Reply:
look at the implementation of this function